test/stdlib/CodableTests.swift

```diff
@@ -462,6 +471,58 @@ class TestCodable : TestCodableSuper {
         }
     }
 
+    func test_Dictionary_JSON() {
+        enum X: String, Codable { case a, b }
+        enum Y: String, Codable, CodingKeyRepresentable { case a, b }
+        enum Z: String, Codable, CodingKeyRepresentable {
+            case a
+            case b
+            init?<T: CodingKey>(codingKey: T) {
+                self.init(rawValue: codingKey.stringValue)
+            }
+            var codingKey: CodingKey {
+                GenericCodingKey(stringValue: self.rawValue)
+            }
+        }
+
+        enum U: Int, Codable { case a = 0, b}
+        enum V: Int, Codable, CodingKeyRepresentable { case a = 0, b }
+        enum W: Int, Codable, CodingKeyRepresentable {
+            case a = 0
+            case b
+            init?<T: CodingKey>(codingKey: T) {
+                self.init(rawValue: codingKey.intValue!)
```

Comment:
Should initialization of `W` fail, rather than force-unwrapping here?

Reply:
Yes, thanks.